### PR TITLE
llvm: Add support for selecting runtime library

### PR DESCRIPTION
**Pull request by duda-patryk**
This patch adds Kconfig options to select either GNU libgcc or LLVM compiler-rt. The `rtlib` flag is provided in a config file, so this patch introduces `clang_libgcc.cfg` and `clang_compiler_rt.cfg` which enable appropriate library. The file is selected by concatenating the `clang_` prefix with library name.

With this PR I'm able to compile hello_world sample for google_dragonclaw board with following changes in config:
```
diff --git a/boards/arm/google_dragonclaw/google_dragonclaw_defconfig b/boards/arm/google_dragonclaw/google_dragonclaw_defconfig
index c39f9c6a26..cf6c660243 100644
--- a/boards/arm/google_dragonclaw/google_dragonclaw_defconfig
+++ b/boards/arm/google_dragonclaw/google_dragonclaw_defconfig
@@ -25,3 +25,7 @@ CONFIG_PINCTRL=y
 # Enable MPU and HW stack protection
 CONFIG_ARM_MPU=y
 CONFIG_HW_STACK_PROTECTION=y
+
+CONFIG_LLVM_USE_LLD=y
+CONFIG_LLVM_COMPILER_RT=y
+CONFIG_BUILD_NO_GAP_FILL=y
```

and using command `ZEPHYR_TOOLCHAIN_VARIANT=llvm west build --pristine -b google_dragonclaw samples/hello_world`

### CODE REVIEW

File: cmake/toolchain/llvm/target.cmake [hunk 1]
```diff
@@ -42,3 +42,12 @@ if(DEFINED triple)
 
   unset(triple)
 endif()
+
+if(CONFIG_LLVM_LIBGCC)
+  set(RUNTIME_LIB "libgcc")
```
Comment (tejlmand) [inline cmake/toolchain/llvm/target.cmake:47]: minor nit, please use lower case naming for variables intended to only be used in this file.
```suggestion
  set(runtime_lib"libgcc")
```

File: cmake/toolchain/llvm/target.cmake [hunk 2]
```diff
@@ -42,3 +42,12 @@ if(DEFINED triple)
...
+elseif(CONFIG_LLVM_COMPILER_RT)
+  set(RUNTIME_LIB "compiler_rt")
```
Comment (tejlmand) [inline cmake/toolchain/llvm/target.cmake:49]:
```suggestion
  set(runtime_lib"compiler_rt")
```

File: cmake/toolchain/llvm/target.cmake [hunk 3]
```diff
@@ -42,3 +42,12 @@ if(DEFINED triple)
...
+endif()
+
+list(APPEND TOOLCHAIN_C_FLAGS --config ${ZEPHYR_BASE}/cmake/toolchain/llvm/clang_${RUNTIME_LIB}.cfg)
+list(APPEND TOOLCHAIN_LD_FLAGS --config ${ZEPHYR_BASE}/cmake/toolchain/llvm/clang_${RUNTIME_LIB}.cfg)
```
Comment (tejlmand) [inline cmake/toolchain/llvm/target.cmake:53]:
```suggestion
list(APPEND TOOLCHAIN_C_FLAGS --config ${ZEPHYR_BASE}/cmake/toolchain/llvm/clang_${runtime_lib}.cfg)
list(APPEND TOOLCHAIN_LD_FLAGS --config ${ZEPHYR_BASE}/cmake/toolchain/llvm/clang_${runtime_lib}.cfg)
```

**Review by tejlmand — COMMENTED** [review]
looks good.
A minor nit noticed.

File: cmake/toolchain/llvm/Kconfig [hunk 4]
```diff
@@ -17,3 +17,20 @@ config LLVM_USE_LLD
 	  Use LLVM built-in lld linker with llvm/clang.
 
 endchoice
+
+choice LLVM_RTLIB
```
Comment (tejlmand) [inline cmake/toolchain/llvm/Kconfig:21]: no changes requested atm, but I think we should consider if selection of runtime library should be made available in a dedicated `lib/runtime/Kconfig` file here: https://github.com/zephyrproject-rtos/zephyr/tree/main/lib

Much like the fact that we have C standard lib selection located together here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/libc/Kconfig

Even for C standard libs only available to certain toolchains, such as arc mwdt, we have C std lib selection in the common file, ref:
https://github.com/zephyrproject-rtos/zephyr/blob/a18761e29861b709b984eca82dc31efe6214fd62/lib/libc/Kconfig#L94-L97

As runtime lib selection is new in Zephyr build config, the current proposed location is fine, but I would like input from @semihalf-duda-patryk  and others, as to what they believe is a logical location.
Currently I think a location close to C standard lib selection is most logical.

Comment (semihalf-duda-patryk) [reply cmake/toolchain/llvm/Kconfig]: Moved `Kconfig` to `lib/runtime`

File: cmake/toolchain/llvm/target.cmake (hunk 3 above)
Comment (semihalf-duda-patryk) [reply cmake/toolchain/llvm/target.cmake]: Fixed.

File: cmake/toolchain/llvm/target.cmake (hunk 2 above)
Comment (semihalf-duda-patryk) [reply cmake/toolchain/llvm/target.cmake]: Fixed.

File: cmake/toolchain/llvm/target.cmake (hunk 1 above)
Comment (semihalf-duda-patryk) [reply cmake/toolchain/llvm/target.cmake]: Fixed.

**Review by tejlmand — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM, this is great work. Being able to use compiler-rt is a major achievement.

**Pull request merged**